02-s3/lab2-2-3/template.yml

```diff
@@ -0,0 +1,52 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: "Creates an S3 bucket for Stelligent-U Topic 1: CF Practice 1.1"
+Parameters:
+  BucketName:
+    Description: The name of the bucket
+    Type: String
+    Default: 'friendly-name'
+  UserName:
+    Description: The name of the user
+    Type: String
+    Default: 'friendly-user-name'
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Ref BucketName
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref BucketName
+      PolicyDocument:
+        Statement:
+          - Action:
+              - 's3:Get*'
+              - 's3:List*'
+            Effect: Allow
+            Resource:
+              - !Join ['', ['arn:aws:s3:::',!Ref BucketName]]
+              - !Join ['', ['arn:aws:s3:::',!Ref BucketName, '/*']]
+            Principal: '*'
+  S3ReadPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties: 
+      Description: Allows read to all S3 buckets and objects within this AWS account
+      ManagedPolicyName: imbgar-S3Read
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - s3:*
+            Resource:
+              - arn:aws:s3:::imbgarlab
+              - arm:aws:s3:::imbgarlab/*
+      Users: 
+        - !Ref UserName
+Outputs:
+  PolicyARN:
+    Description: The ARN of the created IAM policy
+    Value: !Ref S3ReadPolicy
+    Export:
+      Name: !Join [ "-", [ !Ref "AWS::StackName", PolicyARN ] ]
```

Comment:
I wouldn't be prefixing these arns with the stack name, you would need to parse them instead of being able to use them directly, also, you couldn't import them into other stacks.